Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift

```diff
@@ -66,28 +81,67 @@ internal final class AnalyticsProvider: AnalyticsProviderProtocol {
     }
 
     // MARK: - Internal
-    
-    internal func fetchAndCacheCheckoutAttemptIdIfNeeded() {
-        fetchCheckoutAttemptId { _ in /* Do nothing, the point is to trigger the fetching and cache the value  */ }
-    }
 
-    internal func fetchCheckoutAttemptId(completion: @escaping (String?) -> Void) {
-        guard configuration.isEnabled else {
+    internal func fetchCheckoutAttemptId(with flavor: TelemetryFlavor, additionalFields: AdditionalAnalyticsFields?) {
+        guard configuration.isEnabled, configuration.isTelemetryEnabled else {
             checkoutAttemptId = "do-not-track"
-            completion(checkoutAttemptId)
             return
         }
+        if case .dropInComponent = flavor { return }
+        
+        let telemetryData = TelemetryData(flavor: flavor,
+                                          additionalFields: additionalFields,
+                                          context: configuration.context)
 
-        let checkoutAttemptIdRequest = CheckoutAttemptIdRequest()
+        let checkoutAttemptIdRequest = CheckoutAttemptIdRequest(data: telemetryData)
 
         uniqueAssetAPIClient.perform(checkoutAttemptIdRequest) { [weak self] result in
             switch result {
             case let .success(response):
                 self?.checkoutAttemptId = response.identifier
-                completion(response.identifier)
             case .failure:
-                completion(nil)
+                self?.checkoutAttemptId = nil
             }
         }
     }
+    
+    internal func send(event: AdyenAnalytics.Event) {
+        events.append(event)
+    }
+    
+    internal func send(log: AdyenAnalytics.Log) {
+        logs.append(log)
+    }
+    
+    internal func send(error: AdyenAnalytics.Error) {
+        errors.append(error)
+        sendAll()
+    }
+    
+    private func setupTimer() {
+        let timer = Timer(timeInterval: 10, target: self, selector: #selector(sendAll), userInfo: nil, repeats: true)
+        timer.tolerance = 1
+        RunLoop.current.add(timer, forMode: .common)
+    }
+    
+    @objc private func sendAll() {
+        guard configuration.isEnabled,
+              let checkoutAttemptId else { return }
+        var request = AdyenAnalyticsRequest(checkoutAttemptId: checkoutAttemptId)
```

Comment:
Maybe check if there is actual data points to be sent to avoid useless api calls.

Or maybe stop the timer after the api call if there is nothing to send and only start it again after a data point arrives 🤔

Reply:
Think committing this part was not needed as it was not done. I will work on this part on next PR

Reply:
Ahh I see - this code is not even called yet? 👌